client/pages/Messages.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import io from 'socket.io-client';
+
+export default class Messages extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      message: ''
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSend = this.handleSend.bind(this);
+  }
+
+  handleClick() {
+    this.props.history.push('/matches');
+  }
+
+  handleChange(event) {
+    this.setState({
+      message: event.target.value
+    });
+  }
+
+  handleSend(event) {
+    const params = new URLSearchParams(this.props.location.search);
+    const userId = params.get('userId');
+    const partnerId = params.get('partnerId');
+    const socket = io();
```

Comment:
Rather than instantiate a whole new socket for every message, you will want to manage the lifespan of the socket. We do it like this:
```js
componentDidMount() {
  this.socket = io();
}

componentWillUnmount() {
  this.socket.disconnect();
}
```